metrics/memory/analyze_heap_memory.py

```
@@ -0,0 +1,1037 @@
+    g: G
+    c_count: Counter
+    c_self: Counter
+    t_count: Counter
+    t_self: Counter
+    buckets: Dict[str, Any]
+    top_ret: List[Dict[str, Any]]
+    detached: Dict[str, Any]
+    timeline: Dict[str, Any]
+
+
+def analyze_cap(cap: CaptureFile, payload: Dict[str, Any], strict: bool, top_n: int) -> A:
```

Ruff (0.15.1)

[error] Boolean-typed positional argument in function definition

(FBT001)